src/core/rooms.ts

```diff
@@ -103,11 +107,46 @@ export class DefaultRooms implements Rooms {
     this._logger.trace('Rooms.release();', { roomId });
 
     const room = this._rooms.get(roomId);
-    if (!room) return Promise.resolve();
+    const releasing = this._releasing.get(roomId);
+
+    // if we don't have the room
+    if (!room) {
+      // ... and it's currently releasing wait for the same promise
+      if (releasing) {
+        return releasing.promise;
+      }
+
+      // if not releasing, nothing else to do
```

Comment:
Small nitpicks again,
`// if the room is not releasing, there is nothing else to do`